index.js

```diff
@@ -1,40 +1,83 @@
 #!/usr/bin/env node
 require('shelljs/global');
 
-// require npm user
-// bump package version
-// commit
-// create tag
-// push commit & tag
-// publish
+var path = require('path');
+
+// npm version (bump version, commit, make tag)
+// echo npm publish
+//  - if successful, push commit & tag
+//  - if failed, undo commit & tag
 
 function usage() {
-  echo("");
-  echo("  Usage: node " + process.argv[1] + " <major|minor|patch>");
+  echo('');
+  echo('  Usage: node ' + process.argv[1] + ' <major|minor|patch>');
 }
 
 config.silent = true;
 function run(version) {
-  var npm_user = exec('npm whoami').trimRight();
-  var is_collaborator = exec('npm access ls-collaborators').grep('.*' + npm_user + '.*:.*write.*').trimRight();
-  var is_owner = exec('npm owner ls').grep('.*' + npm_user + ' <.*').trimRight();
-
-  if (npm_user) {
-    config.silent = false;
-    if (is_collaborator || is_owner) {
-      echo('Publishing new ' + version + ' version as ' + npm_user + '.');
+  config.silent = false;
+  config.fatal = true;
+  try {
+    // TODO(nfischer): only allow releases from master branch (issue #4)
```

Comment:
This can be checked easily:
```js
if (exec('git rev-parse --abbrev-ref HEAD').trim() !== 'master') {
    echo('');
    echo('Releases are permitted only from the master branch.');
    exit(1);
}
```

Reply:
Let's keep it in a separate PR